### PR TITLE
app-arch/zstd: add threads use flag

### PR DESCRIPTION
**Pull request by atomsymbol**
Please merge. Thanks.

### CODE REVIEW

**Comment by atomsymbol** [comment]
> I don't see any evidence that libarchive links with libzstd-mt, so I don't think this change to libarchive is warranted.

There is no `libzstd-mt.so`, there is just `libzstd.so` that can be built with or without `‑DZSTD_MULTITHREAD`.

The `libzstd-mt` target simply adds `‑DZSTD_MULTITHREAD` to compiler flags and then builds the target `libzstd`.

If a user has (has not) threads in the use flags of libarchive, it is reasonable to assume that the user wants (does not want) libarchive to automatically use threads in case of zstd archives.

    $ ldd /usr/lib64/libarchive.so
        libzstd.so.1 => /usr/lib64/libzstd.so.1

**Comment by mattst88** [comment]
Thanks for the explanation.

Without an actual dependence on some aspect of multithreaded-zstd support libarchive's ebuild shouldn't depend on it, much less with [threads=] so that they have to be built with the same setting. I suspect you copied this from the lzma dep, but libarchive apparently does care about multithread support existing in lzma. configure.ac has:

>     [whether we have multithread support in lzma],

Please drop the libarchive change unless it actually needs to depend on zstd with a particular thread setting.

**Comment by atomsymbol** [comment]
The libarchive change has been dropped.

File: app-arch/zstd/zstd-1.3.7-r1.ebuild [hunk 1]
```diff
@@ -12,10 +12,10 @@ SRC_URI="https://github.com/facebook/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
 LICENSE="|| ( BSD GPL-2 )"
 SLOT="0/1"
 KEYWORDS="~alpha amd64 arm arm64 hppa ia64 m68k ~mips ppc ppc64 s390 sh sparc x86 ~amd64-linux ~x86-linux ~ppc-macos ~x64-macos ~x86-macos ~sparc-solaris ~sparc64-solaris ~x64-solaris ~x86-solaris"
-IUSE="lz4 static-libs"
+IUSE="lz4 static-libs +threads"
 
-RDEPEND="app-arch/xz-utils
-	lz4? ( app-arch/lz4 )"
+RDEPEND="app-arch/xz-utils[threads=,${MULTILIB_USEDEP}]
```
Comment (mattst88) [inline app-arch/zstd/zstd-1.3.7-r1.ebuild]: `-lz -llzma -llz4` is only used in linking `zstd`, which is only done on the native ABI, so the multlib USE deps are not necessary. Neither is the `threads=` USE dep. That should be evident by the fact that if `threads=` were necessary, then zstd *without* threads would not successfully link with xz-utils *with* threads.

**Comment by mattst88** [comment]
Thanks for the contribution! I fixed up the multilib USE deps, and only applied the patch to the unstable zstd and committed it.

**Pull request closed without merging**